src/main/java/snownee/lychee/recipes/ShapedCraftingRecipe.java

```diff
@@ -96,9 +96,14 @@ public IntList getItemIndexes(JsonPointer pointer) {
 			}
 			var data = dataOptional.get();
 			IntList list = IntArrayList.of();
+			int cp = key.codePointAt(0);
+			int index = 0;
 			for (var i = 0; i < data.pattern().size(); i++) {
-				if (key.equals(data.pattern().get(i))) {
-					list.add(i);
+				for (char c : data.pattern().get(i).toCharArray()) {
+					if (c == cp) {
+						list.add(index);
+					}
+					index++;
```

Comment:
```suggestion
                var row = data.pattern().get(i);
				for (var j = 0; j < row.length(); j++) {
					if (row.charAt(j) == cp) {
						list.add(index);
					}
					index++;
```
`toCharArray` 会新建一个数组，直接 `charAt` 吧